driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -714,21 +670,16 @@ public void onException(Connection connection, Exception exception, long latency
             Host queriedHost = current;
             try {
                 connection.release();
-
-                if (exception instanceof ConnectionException) {
-                    RetryPolicy retryPolicy = retryPolicy();
-                    RetryPolicy.RetryDecision decision = retryPolicy.onRequestError(statement, request().consistency(), (ConnectionException) exception, retriesByPolicy);
-                    if (metricsEnabled()) {
-                        metrics().getErrorMetrics().getConnectionErrors().inc();
-                        if (decision.getType() == Type.RETRY)
-                            metrics().getErrorMetrics().getRetriesOnConnectionError().inc();
-                        if (decision.getType() == Type.IGNORE)
-                            metrics().getErrorMetrics().getIgnoresOnConnectionError().inc();
-                    }
-                    processRetryDecision(decision, connection, exception);
-                    return;
+                RetryPolicy retryPolicy = retryPolicy();
+                RetryPolicy.RetryDecision decision = retryPolicy.onRequestError(statement, request().consistency(), (DriverException) exception, retriesByPolicy);
```

Comment:
You will notice the cast to `DriverException`, which is harmless because in practice all exceptions are instances of `DriverException`.